libsnark/gadgetlib1/gadgets/hashes/anemoi/anemoi_components.tcc

```diff
@@ -0,0 +1,528 @@
+/** @file
+ *****************************************************************************
+
+ Implementation of interfaces for top-level Anemoi hash function gadgets.
+
+ See anemoi_gadget.hpp .
+
+ *****************************************************************************
+ * @author     This file is part of libsnark, developed by SCIPR Lab
+ *             and contributors (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#ifndef LIBSNARK_GADGETLIB1_GADGETS_HASHES_ANEMOI_COMPONENTS_TCC_
+#define LIBSNARK_GADGETLIB1_GADGETS_HASHES_ANEMOI_COMPONENTS_TCC_
+
+namespace libsnark
+{
+
+// A R1CS constraint is a formal expression of the form
+//
+//                < A , X > * < B , X > = < C , X > ,
+//
+// where X = (x_0,x_1,...,x_m) is a vector of formal variables and
+// A,B,C each consist of 1+m elements in <FieldT> and <A, X> = \sum_i
+// (a_i x_i) is the dot product between vectors A and X. Equivalently,
+// the vectors A,B,C are linear combinations of X.
+//
+// A R1CS constraint is used to construct a R1CS constraint system.
+//
+// See also class \ef r1cs_constraint
```

Comment:
This seems to belong in the files related to r1cs constraints, if not already explained there.

Reply:
Removed.